spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/fieldvalues/UnresolvableDefaultValue.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.configurationprocessor.fieldvalues;
+
+/**
+ * A placeholder for a field default value used to indicate the field was unable to be
+ * resolved by the parser.
+ *
+ * @author Chris Bono
+ * @since 2.6.0
+ */
+public class UnresolvableDefaultValue {
```

Comment:
I do not ❤️ using this marker interface to signal the condition but its deep enough down the chain that rippling something else to track this fact seemed less desirable than this approach. I do think we can add some default methods in `FieldResolver` to signal that a field was unresolvable and also inspect if a field was unresolvable - that would remove the config processor from having direct knowledge of this inner impl detail.